src/lib/transport.js

```diff
@@ -0,0 +1,290 @@
+'use strict'
+
+const Packet = require('../utils/packet')
+const moment = require('moment')
+const cryptoHelper = require('../utils/crypto')
+const cc = require('../utils/condition')
+const co = require('co')
+const debug = require('debug')('ilp:transport')
+const assert = require('assert')
+const base64url = require('../utils/base64url')
+const BigNumber = require('bignumber.js')
+const { safeConnect } = require('../utils')
+const { createDetails, parseDetails } = require('../utils/details')
+
+function _safeDecrypt (data, secret) {
+  if (!data) return {}
+  try {
+    return parseDetails({ details: data, secret })
+  } catch (err) {
+    debug('decryption error="' + err.message + '"', 'data="' + data + '"')
+    return undefined
+  }
+}
+
+function createPacketAndCondition ({
+  id,
+  destinationAmount,
+  destinationAccount,
+  secret,
+  data,
+  headers,
+  unsafeHeaders,
+  expiresAt
+}, protocol) {
+  assert(typeof destinationAmount === 'string', 'destinationAmount must be a string')
+  assert(typeof destinationAccount === 'string', 'destinationAccount must be a string')
+  assert(Buffer.isBuffer(secret), 'secret must be a buffer')
```

Comment:
I don't think so. This is a low level library, so people using it will invariably have opportunities to do insecure things if they want to. It's more likely to get in the way of people testing or developing things on top of this.